plutus-contract/src/Plutus/Contract/Wallet.hs

```diff
@@ -288,9 +288,12 @@ mkSpendingRedeemers P.Tx{P.txInputs} = fmap join (traverse extract txInputs) whe
     extract _ = pure []
 
 mkMintingRedeemers :: P.Tx -> Either CardanoAPI.ToCardanoError [ExportTxRedeemer]
-mkMintingRedeemers P.Tx{P.txRedeemers, P.txMintScripts} = traverse extract $ Map.toList txRedeemers where
+mkMintingRedeemers P.Tx{P.txRedeemers, P.txMintScripts} =
+    catMaybes <$> traverse extract (Map.toList txRedeemers)
+ where
     indexedMintScriptHashes = Map.fromList $ zip [0..] $ Map.keys txMintScripts
     extract (PV1.RedeemerPtr PV1.Mint idx, redeemer) = do
         redeemerPolicyId <- maybe (Left CardanoAPI.MissingMintingPolicy) Right (Map.lookup idx indexedMintScriptHashes)
-        pure MintingRedeemer{redeemer, redeemerPolicyId}
-    extract (PV1.RedeemerPtr tag _, _) = Left (CardanoAPI.ScriptPurposeNotSupported tag)
+        pure $ Just MintingRedeemer{redeemer, redeemerPolicyId}
+    -- Some other redeemer (like a spending redeemer) which is ignored
+    extract (PV1.RedeemerPtr _ _, _) = pure Nothing
```

Comment:
@ak3n Can you validate this part which I'm not sure?

Running the executable in `plutus-use-cases` which calls `mkMintingRedeemers` resulted in a `ScriptPurposeNotSupported Spend` error. This means that the `txRedeemers` in the `Tx` type now contains redeemers for minting AND spending. Is that okay? I suppose so, but I don't know why we didn't have spending redeemers in `txRedeemers` in the past.

Reply:
As I understand the code, it's okay. The redeemers dictionary contains redeemers for every purpose. And then we analyze the fields for each purpose (minting, spending, etc) and use that dictionary. That's how I read `mkMintingRedeemers` function which uses `txMintScripts`. I suppose throwing the error here is incorrect and the fix makes sense as any type of redeemer could be there.

The interesting question — what triggered this?

Also, at the same time at the declaration of `Tx`:

```
    txRedeemers       :: Redeemers,
    -- ^ Redeemers of the minting scripts.
```

I suppose that we treated the field as a `Mint`-only redeemers?

Reply:
Yeah, that's my though. We only treated this as a `Mint`-only redeemers.